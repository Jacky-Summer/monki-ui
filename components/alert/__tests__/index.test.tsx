import React from 'react'
import { mount } from 'enzyme'
import Alert, { AlertProps } from '../alert'

const testProps: AlertProps = {
  message: 'This is a Alert',
  type: 'success',
  closable: true,
  onClose: jest.fn(),
}

jest.useFakeTimers()

describe('Alert component', () => {
  it('should render the correct default alert', () => {
    const wrapper = mount(
      <Alert
        message="I am ALert"
        description="This is some desc about the Alert"
      />,
    )
    expect(wrapper.find('.mk-alert').hasClass('mk-alert-warning')).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })

  it('should be closed when click the alert close icon', () => {
    const wrapper = mount(<Alert {...testProps} />)
    wrapper.find('.mk-alert-close').simulate('click')
    jest.advanceTimersByTime(1000)
    expect(testProps.onClose).toHaveBeenCalled()
  })
})
