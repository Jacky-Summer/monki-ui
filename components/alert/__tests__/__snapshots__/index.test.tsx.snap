// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component should render the correct default alert 1`] = `
<Alert
  closable={false}
  description="This is some desc about the Alert"
  message="I am ALert"
  type="warning"
>
  <Transition
    animation="zoom-in-left"
    appear={true}
    in={true}
    timeout={300}
    unmountOnExit={true}
    wrapper={false}
  >
    <CSSTransition
      appear={true}
      classNames="zoom-in-left"
      in={true}
      timeout={300}
      unmountOnExit={true}
    >
      <Transition
        appear={true}
        enter={true}
        exit={true}
        in={true}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={300}
        unmountOnExit={true}
      >
        <div
          className="mk-alert className mk-alert-warning"
        >
          <span
            className="mk-alert-message"
          >
            I am ALert
          </span>
          <p
            className="mk-alert-desc"
          >
            This is some desc about the Alert
          </p>
        </div>
      </Transition>
    </CSSTransition>
  </Transition>
</Alert>
`;
